### PR TITLE
Simple RTPS Participant Implementation [13008]

### PR DESCRIPTION
**Pull request by jparisu**
Based on branch #54

### CODE REVIEW

File: test/blackbox/implementations/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+################
+# Implementations Test #
+################
```
Comment (rsanchez15) [inline test/blackbox/implementations/CMakeLists.txt:17]:
```suggestion
####################
# Implementations Test #
####################
```

File: test/blackbox/implementations/ImplementationsTest.cpp [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        // Let DDSRouter object destroy for the next iteration
+    }
+}
+
+/**
+ * Test that creates a DDSRouter with a Pair of Participants of same type.
+ * It creates a DDSRouter with two Participants of same kind, starts it with an active topic,
+ * then stops it and finally destroys it.
+ *
+ * This test will fail if it brokes.
```
Comment (rsanchez15) [inline test/blackbox/implementations/ImplementationsTest.cpp:110]:
```suggestion
 * This test will fail if it crashes.
```

File: test/blackbox/implementations/ImplementationsTest.cpp [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        // Let DDSRouter object destroy for the next iteration
+    }
+}
+
+/**
+ * Test that creates a DDSRouter with several Participants, one of each type
+ * It creates a DDSRouter with a Participant of each kind,
+ * starts it with an active topic, then stops it and finally destroys it.
+ *
+ * This test will fail if it brokes.
```
Comment (rsanchez15) [inline test/blackbox/implementations/ImplementationsTest.cpp:145]:
```suggestion
 * This test will fail if it crashes.
```

File: test/blackbox/implementations/ImplementationsTest.cpp [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+ */
+TEST(ImplementationsTest, all_implementations)
+{
+    // Activate log
+    Log::SetVerbosity(Log::Kind::Info);
+    Log::SetCategoryFilter(std::regex("(DDSROUTER)"));
```
Comment (rsanchez15) [inline test/blackbox/implementations/ImplementationsTest.cpp:151]: Remove log traces

File: test/blackbox/implementations/ImplementationsTest.cpp [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+}
+
+int main(
+        int argc,
+        char** argv)
+{
+    // Activate log
+    Log::SetVerbosity(Log::Kind::Info);
+    Log::SetCategoryFilter(std::regex("(DDSROUTER)"));
+
```
Comment (rsanchez15) [inline test/blackbox/implementations/ImplementationsTest.cpp:192]: Remove log traces

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: include/ddsrouter/participant/implementations/rtps/SimpleRTPSRouterParticipant.hpp [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+#include <ddsrouter/configuration/SimpleRTPSParticipantConfiguration.hpp>
+#include <ddsrouter/participant/implementations/auxiliar/BaseParticipant.hpp>
+#include <ddsrouter/reader/implementations/rtps/RTPSRouterReader.hpp>
+#include <ddsrouter/writer/implementations/rtps/RTPSRouterWriter.hpp>
+
+namespace eprosima {
+namespace ddsrouter {
+
+/**
+ * Concrete Participant with Simple Discovery Protocol.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/participant/implementations/rtps/SimpleRTPSRouterParticipant.hpp:35]: :+1:

File: include/ddsrouter/participant/implementations/auxiliar/impl/BaseParticipant.ipp [hunk 7]
```diff
@@ -14,25 +14,29 @@
 
 /**
  * @file BaseParticipant.ipp
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/participant/implementations/auxiliar/impl/BaseParticipant.ipp:16]: Deprecated comment

File: src/cpp/configuration/SimpleRTPSParticipantConfiguration.cpp [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+namespace ddsrouter {
+
+// SimpleRTPSParticipantConfiguration& SimpleRTPSParticipantConfiguration::operator =(
+//         const ParticipantConfiguration& other) noexcept
+// {
+//     this->id_ = other.id_;
+//     this->type_ = other.type_;
+//     this->raw_configuration_ = other.raw_configuration_;
+//     return *this;
+// }
```
[the remark this answers is not in the document]
Comment (jparisu) [reply src/cpp/configuration/SimpleRTPSParticipantConfiguration.cpp:34]: Nope

File: src/cpp/main.cpp [hunk 9]
```diff
@@ -26,5 +35,24 @@ int main(
     static_cast<void>(argc);
     static_cast<void>(argv);
 
+    // Activate log
+    Log::SetVerbosity(Log::Kind::Info);
```
[the remark this answers is not in the document]
Comment (jparisu) [reply src/cpp/main.cpp:39]: :+1:

File: test/blackbox/implementations/ImplementationsTest.cpp (hunk 4 above)
Comment (jparisu) [reply test/blackbox/implementations/ImplementationsTest.cpp:151]: :+1:

File: include/ddsrouter/participant/implementations/rtps/SimpleRTPSRouterParticipant.hpp [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+ *
+ * Standard RTPS Participant with Simple Discovery and default attributes.
+ */
+class SimpleRTPSRouterParticipant : public BaseParticipant<SimpleRTPSParticipantConfiguration>
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/participant/implementations/rtps/SimpleRTPSRouterParticipant.hpp]: Solve in commit 8724e63

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**